src/__tests__/cogniteTimeSeriesSearch.test.ts

```diff
@@ -0,0 +1,226 @@
+      expect(searchRequest.view.version).toBe('v1');
+      expect(searchRequest.query).toBe('temperature');
+      expect(searchRequest.limit).toBe(10);
+      expect(searchRequest.filter?.not?.equals?.property).toEqual(['type']);
+      expect(searchRequest.filter?.not?.equals?.value).toBe('string');
+    });
+
+    it('should support empty query for browsing all instances', () => {
+      const browseRequest = {
+        view: {
+          type: 'view' as const,
+          space: 'cdf_cdm',
+          externalId: 'CogniteTimeSeries',
+          version: 'v1'
+        },
+        limit: 1000
+      };
+
+      expect(browseRequest.view.type).toBe('view');
+      expect(browseRequest.limit).toBe(1000);
+      expect('query' in browseRequest).toBe(false);
+    });
+  });
+
+  describe('Tab Configuration', () => {
+    it('should have proper tab properties', () => {
+      // Test the tab enum value
+      expect(Tab.CogniteTimeSeriesSearch).toBe('CogniteTimeSeries');
+      
+      // Verify it's a string type (important for tab routing)
+      expect(typeof Tab.CogniteTimeSeriesSearch).toBe('string');
+    });
+
+    it('should support standard timeseries options', () => {
+      const baseQuery = {
+        ...defaultCogniteTimeSeriesSearchQuery,
+        selectedTimeseries: {
+          space: 'cdf_cdm',
+          externalId: 'ts1',
+          name: 'Test Sensor'
+        }
+      };
+
+      // Should be compatible with standard timeseries options
+      const extendedQuery = {
+        ...baseQuery,
+        aggregation: 'average',
+        granularity: '1h',
+        latestValue: false
+      };
+
+      expect(extendedQuery.aggregation).toBe('average');
+      expect(extendedQuery.granularity).toBe('1h');
+      expect(extendedQuery.latestValue).toBe(false);
+    });
+  });
+
+  describe('Label Generation', () => {
+    it('should prefer selectedTimeseries name for labels', () => {
+      const query: CogniteTimeSeriesSearchQuery = {
+        space: 'cdf_cdm',
+        version: 'v1',
+        externalId: 'CogniteTimeSeries',
+        selectedTimeseries: {
+          space: 'cdf_cdm',
+          externalId: 'temp_sensor_01',
+          name: 'Building A - Temperature Sensor'
+        }
+      };
+
+      // The label should prioritize the name field
+      const expectedLabel = query.selectedTimeseries?.name;
+      expect(expectedLabel).toBe('Building A - Temperature Sensor');
+    });
+
+    it('should fallback to instanceId format when name is not available', () => {
+      const query: CogniteTimeSeriesSearchQuery = {
+        space: 'cdf_cdm',
+        version: 'v1',
+        externalId: 'CogniteTimeSeries',
+        selectedTimeseries: {
+          space: 'cdf_cdm',
+          externalId: 'temp_sensor_01'
+          // No name field
+        }
+      };
+
+      // Should fallback to space:externalId format
+      const fallbackLabel = `${query.selectedTimeseries?.space}:${query.selectedTimeseries?.externalId}`;
```

Comment:
tests in this file don't actually test anything.
best would be to have both an e2e (playwright) and a unit test with mocked http calls.
we have examples of both of these in the repo